multicluster/cmd/gateways.go

```diff
@@ -1,95 +1,218 @@
 			if err != nil {
-				fmt.Fprintln(os.Stderr, err)
+				fmt.Fprintf(os.Stderr, "failed to list pods in namespace %s: %s", multiclusterNs.Name, err)
 				os.Exit(1)
 			}
 
-			renderGateways(resp.GetOk().GatewaysTable.Rows, stdout)
+			var statuses []gatewayStatus
+			gatewayMetrics := getGatewayMetrics(k8sAPI, pods.Items, opts.wait)
+			for _, gateway := range gatewayMetrics {
+				if gateway.err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to get gateway status for %s: %s\n", gateway.clusterName, gateway.err)
+					continue
+				}
+				gatewayStatus := gatewayStatus{
+					clusterName: gateway.clusterName,
+				}
+
+				// Parse the gateway metrics so that we can extract liveness
+				// and latency information.
+				var metricsParser expfmt.TextParser
+				parsedMetrics, err := metricsParser.TextToMetricFamilies(bytes.NewReader(gateway.metrics))
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to parse metrics for %s: %s\n", gateway.clusterName, err)
+					continue
+				}
+
+				// Check if the gateway is alive by using the gateway_alive
+				// metric and ensuring the label matches the target cluster.
+				for _, metrics := range parsedMetrics["gateway_alive"].GetMetric() {
+					if !isTargetClusterMetric(metrics, gateway.clusterName) {
+						continue
+					}
+					if metrics.GetGauge().GetValue() == 1 {
+						gatewayStatus.alive = true
+						break
+					}
+				}
+
+				// Search the local cluster for mirror services that are
+				// mirrored from the target cluster.
+				selector := fmt.Sprintf("%s=%s,%s=%s",
+					k8s.MirroredResourceLabel, "true",
+					k8s.RemoteClusterNameLabel, gateway.clusterName,
+				)
+				services, err := k8sAPI.CoreV1().Services(corev1.NamespaceAll).List(context.Background(), metav1.ListOptions{LabelSelector: selector})
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to list services for %s: %s\n", gateway.clusterName, err)
+					continue
+				}
+				gatewayStatus.numberOfServices = len(services.Items)
+
+				// Check the last observed latency by using the
+				// gateway_latency metric and ensuring the label the target
+				// cluster.
+				for _, metrics := range parsedMetrics["gateway_latency"].GetMetric() {
+					if !isTargetClusterMetric(metrics, gateway.clusterName) {
+						continue
+					}
+					gatewayStatus.latency = uint64(metrics.GetGauge().GetValue())
+					break
+				}
+
+				statuses = append(statuses, gatewayStatus)
+			}
+			renderGateways(statuses, stdout)
 			return nil
 		},
 	}
 
 	cmd.Flags().StringVar(&opts.clusterName, "cluster-name", "", "the name of the target cluster")
-	cmd.Flags().StringVar(&opts.gatewayNamespace, "gateway-namespace", "", "the namespace in which the gateway resides on the target cluster")
-	cmd.Flags().StringVarP(&opts.timeWindow, "time-window", "t", "1m", "Time window (for example: \"15s\", \"1m\", \"10m\", \"1h\"). Needs to be at least 15s.")
+	cmd.Flags().DurationVarP(&opts.wait, "wait", "w", opts.wait, "Time allowed to fetch diagnostics")
```

Comment:
tioli: the two help messages for the flags have different casing. I'd go with lowercase or uppercase for the first letter of the sentence (i.e "The name of the target cluster").